mm2src/mm2_core/src/mm_ctx.rs

```diff
@@ -393,23 +387,15 @@ impl MmArc {
 
     pub fn stop(&self) -> Result<(), String> {
         try_s!(self.stop.pin(true));
-        for handler in self.abort_handlers.lock().unwrap().drain(..) {
-            handler.abort();
-        }
+
+        // Notify shutdown listeners.
+        self.graceful_shutdown_registry.abort_all();
+        // Abort spawned futures.
+        self.abortable_system.abort_all();
 
         #[cfg(not(target_arch = "wasm32"))]
         self.background_processors.lock().unwrap().drain();
```

Comment:
I guess background processors can be moved back to `LightningCoin` as before since they will be dropped now with the coins on calling stop, I added this to the lightning integration checklist https://github.com/KomodoPlatform/atomicDEX-API/issues/1045#issue-971642359